server/tweet/test/tweet.test.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-env node, mocha */
+'use strict'
+
+const tweetPlugin = require('../index')
+
+const assert = require('assert')
+const nock = require('nock')
+const MongoClient = require('mongodb').MongoClient
+const Fastify = require('fastify')
+const fp = require('fastify-plugin')
+
+const makeRequest = (fastify, options) => new Promise((resolve) => fastify.inject(options, resolve))
+
+const configuration = {
+  TWEET_MONGO_URL: 'mongodb://localhost/tweet'
+}
+
+let fastify
+describe('tweet', () => {
```

Comment:
I wouldn't use Mocha or put tests within the tweet folder.

Reply:
I like a lot that the tests sit in different folder because the tests are related to that plugin.

Are you preferring only one test folder?

Reply:
one test folder for each package.json file.
So it's very easy to know which code is for prod and which one is for testing.

You can use something like https://github.com/lerna/lerna to have multiple package.json files, but I think it's an overkill for this example.